R/FilterState.R

```diff
@@ -249,12 +249,21 @@ FilterState <- R6::R6Class( # nolint
             `data-bs-toggle` = "collapse",
             href = paste0("#", ns("body")),
             # header elements
-            if (private$is_locked()) icon("lock") else NULL,
-            if (private$is_fixed()) icon("burst") else NULL,
+            if (private$is_anchored()) {
+              if (private$is_fixed()) {
+                icon("anchor-lock")
+              } else {
+                icon("anchor")
+              }
+            } else if (private$is_fixed()) {
+              icon("lock")
```

Comment:
This logic is not wrong but I wonder if it would be easier to read if were more explicit.
```
if (private$is_anchored() && private$is_fixed()) {
  icon("anchor-lock")
} else if (private$is_anchored() && !private$is_fixed()) {
  icon("anchor")
} else if (!private$is_anchored() && private$is_fixed()) {
  icon("lock")
}
```

What do you think?

Reply:
yeah, I agree your version explicitly checks each combination of conditions separately, making it clear which icon will be displayed based on the state of private$is_anchored() and private$is_fixed(). This is better for code maintenance and readability,